func/omp/nested_levels_test.cpp

```diff
@@ -1,236 +1,191 @@
 /**
- *  Checks a few invariants of nested levels.
```

Comment:
This was really complicated and hard to follow, plus it didn't actually pass even on native openmp for me, so I've cut it down to be simpler